packages/element-web-module-api/src/api/custom-components.ts

```diff
@@ -0,0 +1,115 @@
+import type { JSX } from "react";
+import type { RoomEvent } from "matrix-js-sdk";
+
+/**
+ * Targets in Element for custom components.
+ * @public
+ */
+export enum CustomComponentTarget {
+    /**
+     * Component that renders "m.room.message" events in the room timeline.
+     */
+    TextualBody = "TextualBody",
+    /**
+     * "Options" Context menu for a timeline event.
+     * Use `buildContextMenuBlock` to build a section to be used by this component.
+     * @see buildContextMenuBlock
+     */
+    MessageContextMenu = "MessageContextMenu",
```

Comment:
This might be better off as it's own API rather than a custom component. The caller really needs to have a way to build a menu structure which means they need access to the correct Element components. This is provided by `buildContextMenuBlock` but the shape feels wrong.